posts/centralized-rules-execution-server.md

```diff
@@ -0,0 +1,100 @@
+---
+authors:
+- Mallik Sambaraju
+- Amine Ouali Alami
+tags:
+- Rules
+- Kie
+- Drools
+- JHipster
+- AWS ECS
+- Spring Boot
+date: 2010-01-23T10:00:00.000Z
+title: Centralized Rules Engine on AWS ECS
+---
+
+## Introduction
+As software engineers we architect and develop many real world applications solving many business problems, and in those applications we add business logic to solve various issues. Often this business logic is difficult to implement in application code, business rules keep on changing and hard to maintain.
+
+## Why rules engine
+A rules engine may be handy in certain cases where the business logic keeps on changing, there is no easy solution to implement the logic in the code, the code becomes cluttered or may be the business analysts want an easy way to maintain the business rules. Apart from these, rules engines implement rules execution algorithms such as Rete algorithm which are more efficient than traditional if..else or switch statements.
+
+## The problem
+At one of our recent customer's location, we were tasked to resolve a problem and the application has to look into more than hundred different scenarios to come up with a result. Also the customer wanted to have a provision to maintain the rules separately as the rules can change quite often so that the main applications do not have to be redployed whenever there is a rule change.
```

Comment:
```suggestion
At one of our recent customer's locations, we were tasked to resolve a problem, and the application has to look into more than a hundred different scenarios to come up with a result. Also, the customer wanted to have a provision to maintain the rules separately as the rules can change quite often so that the main applications do not have to be redeployed whenever there is a rule change.
```